macaroonbakery/macaroon.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+import copy
+import log
+import os
+
+import bakery
+import codec
+import pymacaroons
+
+MACAROON_V1, MACAROON_V2 = 1, 2
+
+
+class Macaroon:
+    '''Represent an undischarged macaroon along its first
+    party caveat namespace and associated third party caveat information
+    which should be passed to the third party when discharging a caveat.
+    '''
+    def __init__(self, root_key, id, location=None,
+                 version=bakery.LATEST_BAKERY_VERSION, ns=None):
+        '''Create a new macaroon with the given root key, id and location.
+
+        If the version is more than the latest known version,
+        the latest known version will be used. The namespace is that
```

Comment:
I know this was wrong in the original comment, but I think "The namespace is that
of the service creating it" is hard to understand. Perhaps "The namespace
should hold the namespace of the service that is creating the macaroon" ?